src/stdlib/SDL_string.c

```diff
@@ -374,42 +374,76 @@ static size_t SDL_ScanLong(const char *text, int count, int radix, long *valuep)
 static size_t SDL_ScanLongW(const wchar_t *text, int count, int radix, long *valuep)
 {
     const wchar_t *textstart = text;
-    long value = 0;
+    unsigned long value = 0;
     bool negative = false;
 
-    if (*text == '-') {
-        negative = true;
+    if (radix != 0 && (radix < 2 || radix > 36)) {
+        return 0;
+    }
+
+    // don't use SDL_isspace here
+    // wchar_t could in theory be wider than int, which could make the conversion truncate bits
```

Comment:
I don't think that's true for any platform. It's probably worth compile-time asserting and using the SDL_is* functions here.